### PR TITLE
httpclientservice: do not use urljoin

### PR DESCRIPTION
**Pull request by tardyp**
urljoin is stripping the last path if it does not with /
we rather join manually, and enforce baseurl does not end with / and ep start with /

### CODE REVIEW

**Comment by tardyp** [comment]
I think this is good to go. This is serious issue.  Shall we make a 0.9.0p1 for that?

**Comment by seankelly** [comment]
Maybe add in #2507 too.

File: master/buildbot/test/fake/httpclientservice.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class HTTPClientService(service.SharedService):
 
     def __init__(self, base_url, auth=None, headers=None):
         service.SharedService.__init__(self)
+        assert not base_url.endswith("/"), "baseurl should not end with /"
```
Comment (sa2ajj) [inline master/buildbot/test/fake/httpclientservice.py]: I'd rather put it *before* `SharedService.__init__` call: why do extra if things are bad already?

File: master/buildbot/util/httpclientservice.py [hunk 2]
```diff
@@ -90,6 +89,7 @@ class HTTPClientService(service.SharedService):
 
     def __init__(self, base_url, auth=None, headers=None):
         service.SharedService.__init__(self)
+        assert not base_url.endswith("/"), "baseurl should not end with /: " + base_url
```
Comment (sa2ajj) [inline master/buildbot/util/httpclientservice.py]: ditto

**Comment by tardyp** [comment]
@sa2ajj please approve the change, so that we can merge it for 0.9.2

**Pull request merged**